src/main/groovy/com/palantir/gradle/shadowjar/ShadowJarPlugin.java

```diff
@@ -111,6 +112,13 @@ private void setupShadowJarToShadeTheCorrectDependencies(
             conf.setCanBeResolved(false);
         });
 
+        // Tell the consistent versions plugin not to look at the unshaded configuration for the
+        // checkUnusedConstraints task,  It forces resolution before the configuration below is complete which causes
+        // an error under gradle8.
+        VersionRecommendationsExtension recExt =
+                project.getExtensions().getByType(VersionRecommendationsExtension.class);
+        recExt.excludeConfigurations("unshaded");
```

Comment:
Isn't this what we're doing to `rejectedFromShading` below (calling `ShadowJarVersionLock#excludeConfigurationFromVersionsPropsInjection`?)

Reply:
heh - ya.  I did it the "non-groovy" way to experiment with something and then didn't change it back.